### DISCUSSION TITLE
Format response data for text/plain

### DISCUSSION THREAD

**Discussion by Jakub27blue** [dotnet/AspNetCore.Docs#25974] [General]
Hello everyone,
I want to set the [Produces ("text/plain")] attribute for a method that returns an int value. 
Unfortunately, the method returns a status code of 406.
**Code:**
```
        [Produces("text/plain")]
        public int Get()
        {
            return 1;
        }
```
**Request:**
without accept header
**Response:**
Error: response status is 406

In the [documentation](https://docs.microsoft.com/en-us/aspnet/core/web-api/advanced/formatting?view=aspnetcore-6.0):

> By default, ASP.NET Core supports the following media types:
>application/json
>text/json
>text/plain

The documentation is not clear about this.
Do you know how to return an int value in text/plain format?

**Comment by alierguc** [comment]
Hi my friend,

try this method.

```
[HttpGet("GetById")]
public IActionResult GetById()
{
    int returnedValue = 1;

    return Ok(returnedValue);
}
```

**Reply by Jakub27blue** [reply]
I am currently using this solution.
I wonder why there are no built-in formatters programs for built-in types like int or double.

**Reply by alierguc** [reply]
Because what you want to do is to send an http request to a method in the Controller class and return the specified constant value. This is what it should do for the api part.

**Reply by Jakub27blue** [reply]
I mean the formatting of the response type...
The only solution is to write your own formatting program for built-in types besides String.
```
[Produces(MediaTypeNames.Application.Json)]    ///Response: 1
[Produces(MediaTypeNames.Application.Xml)]    ///Response: <int>1</int>
[Produces(MediaTypeNames.Text.Plain)]    //Response: Error: response status is 406
public int Get()
{
     return 1;
}
```
The documentation needs to be corrected.
In my opinion, it is better to return specific type than IActionResult when the values are constant.

**Reply by alierguc** [reply]
Yes, definitely. This method like better explanation to your requests.
